test/src/e2e_vm_tests/test_programs/should_pass/language/ref_mutable_fn_args_bool/src/main.sw

```diff
@@ -0,0 +1,11 @@
+script;
+
+fn mut_arg(ref mut b: bool) {
+    b = true;
+}
+
+fn main() -> bool {
+    let ref mut b = false;
```

Comment:
Hmm I thought we didn't want the `ref` keyword to be allowed anywhere other than function/method parameters. This example would basically be:
```rust
fn mut_arg(ref mut b: bool) {
    b = true;
}

fn main() -> bool {
    let mut b = false;
    mut_arg(b);
    b
}
```
Having to define the variable as `ref` has a lot of implications and I'm not sure we want to go down that rabbit hole 🤔